saleor/graphql/product/bulk_mutations/product_variant_bulk_update.py

```diff
@@ -289,7 +289,30 @@ def clean_stocks(
         variant_index,
         index_error_map,
     ):
+        used_warehouses = defaultdict(list)
+        for stock in stock_global_id_to_instance_map.values():
+            warehouse_global_id = graphene.Node.to_global_id(
+                "Warehouse", stock.warehouse_id
+            )
+            used_warehouses[warehouse_global_id].append(stock.product_variant_id)
+
         if stocks_data := cleaned_input["stocks"].get("create"):
+            for stock_index, stock in enumerate(stocks_data):
+                if (
+                    stock["warehouse"] in used_warehouses
+                    and cleaned_input["id"].id in used_warehouses[stock["warehouse"]]
```

Comment:
Is it `cleaned_input["id"].id` and not `cleaned_input["id"]`?

Reply:
I might be wrong, but I think you can simplify the condition:
```python
if cleaned_input["id"] in used_warehouses.get(stock["warehouse"], {}):
```

Reply:
@SzymJ Btw if tests didn't catch the `cleaned_input["id"].id` issue I assume some paths must have been not covered 🤔